src/mode.c

```diff
@@ -20,10 +21,23 @@ void init_mode(void)
   _armed_state = DISARMED;
 }
 
-void arm(void)
+bool arm(void)
 {
-  _armed_state = ARMED;
-  LED1_ON;
+  static bool started_gyro_calibration = false;
+  if (!started_gyro_calibration && _armed_state == DISARMED)
+  {
+    start_gyro_calibration();
```

Comment:
This is hacky.  I think once we move to a better state machine then this will work better.  For now, I just wanted to run a gyro calibration on arm, so the arm() function returns a boolean of whether or not it succeeded.  The logic works fine for now, but I am certain that it will need to change in the future.